modules/gdscript/gdscript.cpp

```diff
@@ -2450,13 +2470,18 @@ Error ResourceFormatSaverGDScript::save(const Ref<Resource> &p_resource, const S
 	ERR_FAIL_COND_V(sqscr.is_null(), ERR_INVALID_PARAMETER);
 
 	String source = sqscr->get_source_code();
+	ResourceUID::ID uid = ResourceSaver::get_resource_id_for_path(p_path, !p_resource->is_built_in());
 
 	{
 		Error err;
 		Ref<FileAccess> file = FileAccess::open(p_path, FileAccess::WRITE, &err);
 
 		ERR_FAIL_COND_V_MSG(err, err, "Cannot save GDScript file '" + p_path + "'.");
 
+		if (uid != ResourceUID::INVALID_ID && !source.begins_with("# uid://")) {
+			file->store_line("# " + ResourceUID::get_singleton()->id_to_text(uid) + " This line is auto-generated. Do not modify it.");
+		}
```

Comment:
I'm not fond of writing this kind of metadata to script files.

This should likely go to some file in `.godot` that can map scripts to UIDs (fs_cache or similar).

Reply:
`.godot` folder can be cleared at any time, so it should be in some non-ignorable file (like `project.godot`, but storing all scripts in there is overkill).

Reply:
Maybe the script editor should hide this metadata. e.g. Strip it on load and prepend it on save. In this case, put the hidden metadata on the last line might be a better choice since it won't not change the actual line number.

Reply:
I thought about that, but if it's first line, you can fetch it quickly. Getting last line is less performant (and it might have visible effect when you have e.g. 200 scripts that need to be scanned for UID).